proposals/NNNN-Dynamic-App-List-Sorting.md

```diff
@@ -0,0 +1,85 @@
+
+ 1. Set the priority (such as 1st, 2nd, 3rd, etc.) to the parameter `priority` of the `application_policies` on the policy table. (OEMs can modify the priorities themselves.)
+ 2. Implement the `app_list_sorting` parameter below in the SDL Core sdl_preloaded_pt.json and the policy table. (OEMs can modify each parameter value of the policy table.)
+  - "appHMIType" : The sorting method by using the `appHMIType_sorting_list`.
+  - "alphabetical" : The sorting method according to alphabetical order.
+  - "appID" : The sorting method according to the `appID` order of the `application_policies` on the policy table.
+ 3. During the first launch, there is no policy table, therefore the app list is searched in the default policy table at first, then can be searched in SDL Core to obtain the `app_list_sorting` information. If the 'app_list_sorting' information is not present, the list is sorted how it was previously sorted.
+ 4. The apps are registered to the SDL Core by using the `RegisterAppInterface`.
+ 5. The sorting order is performed according to the following procedures.
+	1. Prioritize the `priority` parameter of `application_policies` in the policy table.
+	2. Regarding the apps that do not have a priority value in the `priority` parameter, they are prioritized according to the sorting methods which are set by the `sorting_pattern` of the `app_list_sorting` in the policy table.
+ 6. The sorting list is sent to the HMI by using the `UpdateAppList`.
+
+```json
+app_list_sorting: {
+    "sorting_pattern": "appHMIType",
+    "appHMIType_sorting_list": {
+        "NAVIGATION": 1,
+        "MEDIA": 2,
+        "MESSAGE": 3,
+        "INFORMATION": 4,
+        "COMMUNICATION": 5,
+        "SOCIAL": 6,
+        "PROJECTION": 7,
+        "REMOTE_CONTROL": 8,
+        "SYSTEM": 9,
+        "DEFAULT": 10,
+        "BACKGROUND_PROCESS": 11,
+        "TESTING": 12
+    }
+}
+```
+
+By changing the parameters of  `appHMIType_sorting_list` and `priority` created by each OEM in the policy table, OEMs can display the SDL app icon according to their own intended order.
+
+<b>2. Clarification of icon display sequence</b><br>
+Currently, there is no sequence provided that demonstrates how the icon is displayed. Figure 1 shows the sequence in which the process of this proposal is added to the existing sequence from the program code of SDL Core.
+- The black text indicates the existing order described in the program code.
+- The red text indicates the process added by this proposal.
+
+<b>Figure 1.</b> SDL App Icon display sequence
+![Figure1_SDL_App_Icon_display_sequence.PNG](../assets/proposals/NNNN-Dynamic-App-List-Sorting/Figure1_SDL_App_Icon_display_sequence.PNG)
+
+
+## Potential downsides
+
+As there is a large modification required to implement multiple sorting methods, the developer must do the implementation carefully and it will take a long time. Also, it may need to remove the sorting function from the HMI.
```

Comment:
```suggestion
As there is a large modification required to implement multiple sorting methods, the developer must do the implementation carefully and it will take a long time. Also, the system may need to remove the sorting function from the HMI.
```